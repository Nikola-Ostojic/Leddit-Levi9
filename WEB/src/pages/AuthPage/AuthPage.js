import React, { useState, memo } from "react";
import { Redirect } from "react-router-dom";
import { Container, Button, Box } from "@material-ui/core";

import Login from "./Login";
import Register from "./Register";

export const Auth = ({ user }) => {
  const [showLogin, setShowLogin] = useState(true);

  if (user) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <Box m={10}>
        <Container>
          {showLogin ? <Login /> : <Register />}
          <br />
          <Button
            onClick={() => setShowLogin(!showLogin)}
            variant="contained"
            color="primary"
          >
            {showLogin
              ? "Don't have an account? Register."
              : "Already have an account? Login."}
          </Button>
        </Container>
      </Box>
    </>
  );
};

export default memo(
  Auth,
  ({ user: previousUser }, { user: nextUser }) => previousUser === nextUser
);
