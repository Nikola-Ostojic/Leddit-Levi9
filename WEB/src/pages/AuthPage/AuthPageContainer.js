import { connect } from "react-redux";

import { getUsername } from "modules/Auth/AuthSelectors";

import Auth from "./AuthPage";

export const mapStateToProps = (state) => ({
  user: getUsername(state)
});

export default connect(mapStateToProps)(Auth);
