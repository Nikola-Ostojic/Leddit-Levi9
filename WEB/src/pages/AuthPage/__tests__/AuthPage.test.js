import React from "react";
import { shallow } from "enzyme";
import { Button } from "@material-ui/core";

import { Auth } from "../AuthPage";

// Note: enzyme doesn't support testing functional stateful components using useState hooks
describe("AuthPage test", () => {
  it("Should match snapshot when showLogin is true", () => {
    const wrapper = shallow(<Auth />);

    const btnTxt = wrapper.find(Button).text();

    expect(btnTxt).toEqual("Don't have an account? Register.");
    expect(wrapper).toMatchSnapshot();
  });

  it("Should match snapshot when showLogin is false", () => {
    const wrapper = shallow(<Auth />);

    const btnTxtBeforeSwitchingToRegistration = wrapper.find(Button).text();
    expect(btnTxtBeforeSwitchingToRegistration).toEqual(
      "Don't have an account? Register."
    );

    wrapper.find(Button).simulate("click");

    const btnTxtAfterSwitchingToRegistration = wrapper.find(Button).text();
    expect(btnTxtAfterSwitchingToRegistration).toEqual(
      "Already have an account? Login."
    );

    expect(wrapper).toMatchSnapshot();
  });

  // it("Should match snapshot when the user is already logged in", () => {
  //   const user = "TestUser";
  //   const wrapper = shallow(<Auth user={user} />);

  //   const message = wrapper.find("p").text();

  //   expect(message).toEqual(`You are already logged in ${user}`);
  //   expect(wrapper).toMatchSnapshot();
  // });
});
