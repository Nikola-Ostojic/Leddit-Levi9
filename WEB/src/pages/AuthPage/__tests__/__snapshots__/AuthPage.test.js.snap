// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage test Should match snapshot when showLogin is false 1`] = `
<Fragment>
  <Styled(MuiBox)
    m={10}
  >
    <WithStyles(ForwardRef(Container))>
      <Connect(Register) />
      <br />
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Already have an account? Login.
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Container))>
  </Styled(MuiBox)>
</Fragment>
`;

exports[`AuthPage test Should match snapshot when showLogin is true 1`] = `
<Fragment>
  <Styled(MuiBox)
    m={10}
  >
    <WithStyles(ForwardRef(Container))>
      <Connect(withRouter(Login)) />
      <br />
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Don't have an account? Register.
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Container))>
  </Styled(MuiBox)>
</Fragment>
`;
